### PR TITLE
[Data Layer] Inconsistent ResourceTypes for navigation items

### LINKED ISSUE #1182: [Data Layer] Inconsistent ResourceTypes for navigation items
## Bug Report

**Current Behavior**
When the data layer is enabled for out of the box Core Components some of the `@type` are unexpected. This makes it difficult to filter events based on resource type. For example:

**Breadcrumb Item**:

```
breadcrumb-b7e3652ecf-item-574994a074:
@type: "wknd/components/page"
dc:title: "Climbing New Zealand"
parentId: "breadcrumb-b7e3652ecf"
repo:modifyDate: "2020-01-10T08:54:42Z"
xdm:linkURL: "/content/wknd/us/en/adventures/climbing-new-zealand.html"
```

**Language Navigation link**:

```
languagenavigation-c35513ec9b-item-c0b9aff709:
@type: "wknd/components/page"
dc:title: "Deutsch"
parentId: "page-574994a074"
repo:modifyDate: "2020-01-10T08:54:43Z"
xdm:language: "de-DE"
xdm:linkURL: "/content/wknd/de/de.html"
```

Should the type really be page? I would have expected it to be `wknd/components/breadcrumb` and `wknd/components/languagenavigation`...

**Teaser CTA button**:

```
teaser-da32481ec8-cta-adf3c09db9:
@type: "nt:unstructured"
dc:title: "Surf's Up"
parentId: "teaser-da32481ec8"
xdm:linkURL: "/content/wknd/us/en/magazine/san-diego-surf.html"
```
Type `nt:unstructured` is also unexpected...

**Expected behavior/code**
Items that are listed separately in the data layer but originate from a component should have the resource type for that component, otherwise it is difficult to filter events or understand the component that the event came from...

**Environment**
1. Navigate to https://wknd.site/us/en.html 
1. Open browser console and run `adobeDataLayer.getState()`

### PR DESCRIPTION
**Pull request by bpauli**
- pass parent component to navigation items and generate type for the data layer with the RT of the component + /item

fixes #1182

### CODE REVIEW

File: bundles/core/src/test/resources/teaser/exporter-teaser11.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
       "dataLayer": {
         "teaser-84f85e5d32-cta-4a426bc86e": {
           "xdm:linkURL": "/core/content/teasers.html",
-          "@type": "nt:unstructured",
+          "@type": "core/wcm/components/teaser/v1/teaser/item",
```
Comment (jckautzmann) [inline bundles/core/src/test/resources/teaser/exporter-teaser11.json:16]: @gabrielwalt Do we want the CTA type to be `<teaser-resource-type>/item` or `<teaser-resource-type>/cta`?

Comment (gabrielwalt) [reply bundles/core/src/test/resources/teaser/exporter-teaser11.json:16]: `cta` is more self-explanatory, that might be better.

Comment (bpauli) [reply bundles/core/src/test/resources/teaser/exporter-teaser11.json]: changed teaser action item to /cta in https://github.com/adobe/aem-core-wcm-components/pull/1184/commits/140f7f7543877bb2f33614aef6f672d8ab1663c3

**Review by msagolj — APPROVED** [review]

File: bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/CarouselImplTest.java [hunk 2]
```diff
@@ -64,11 +68,11 @@ void testEmptyCarousel() {
     void testCarouselWithItems() {
         Carousel carousel = getCarouselUnderTest(CAROUSEL_1);
         Object[][] expectedItems = {
-                { "item_1", "Teaser 1", "core/wcm/components/teaser/v1/teaser",
+                { "item_1", "Teaser 1", "cq:Component/item",
```
Comment (jckautzmann) [inline bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/CarouselImplTest.java:71]: shouldn't it be core/wcm/components/teaser/v1/teaser/cta?

Comment (msagolj) [reply bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/CarouselImplTest.java:71]: @jckautzmann this is testing the carousel, there it's /item, no ? Teaser just happend to be the old entry for  @type in that carousel item.

**Review by jckautzmann — APPROVED** [review]

**Pull request merged**